src/ardrive.ts

```diff
@@ -77,6 +116,57 @@ export class ArDriveAnonymous extends ArDriveType {
 		const driveEntity = await this.arFsDao.getPublicDrive(driveId);
 		return Promise.resolve(driveEntity);
 	}
+
+	async getPublicFolder(folderId: string): Promise<ArFSPublicFolder> {
+		const folder = await this.arFsDao.getPublicFolder(folderId);
+		return folder;
+	}
+
+	async getChildrenOfPublicFolder(folderId: FolderID): Promise<ArFSPublicFileOrFolderData[]> {
+		const folder = await this.arFsDao.getPublicFolder(folderId);
+
+		// Fetch all of the folder entities within the drive
+		const driveIdOfFolder = folder.driveId;
+		const allFolderEntitiesOfDrive = (await this.arFsDao.getAllFoldersOfPublicDrive(driveIdOfFolder)).filter(
+			lastFolderRevisionFilter
+		);
+
+		// Feed entities to FolderHierarchy.setupNodesWithEntity()
+		const hierarchy = FolderHierarchy.newFromEntities(allFolderEntitiesOfDrive);
+		const folderIDs = hierarchy.allFolderIDs();
+
+		// Fetch all file entities within all Folders of the drive
```

Comment:
Why do we need ALL the entities? Don't we only need entities with parentFolderIds matching those in the subtree of folderId?

Reply:
You are right. It's like so: 24761eb486d7a9000c1191f8ed13258e16efc9ca

Reply:
This is good but you need to do it for the private folders too.